apps/src/music/progress/ConditionsChecker.ts

```diff
@@ -0,0 +1,46 @@
+// A small helper class that accumulates satisfied conditions, and then evaluates
+// whether a set of conditions have all been satisfied.  Unknown conditions are
+// skipped.  The accumulated satisfied conditions can be cleared at any time.
+
+export default class ConditionsChecker {
+  private currentSatisfiedConditions: string[];
+  private knownConditions: KnownConditions;
+
+  constructor(knownConditions: KnownConditions) {
+    this.currentSatisfiedConditions = [];
+    this.knownConditions = knownConditions;
+  }
+
+  // Reset the accumulated conditions.
+  clear() {
+    this.currentSatisfiedConditions = [];
+  }
+
+  // Accumulate a satisfied condition.
+  addSatisfiedCondition(id: string) {
+    this.currentSatisfiedConditions.push(id);
+  }
+
+  // Check whether the current set of satisfied conditions satisfy the given
+  // required conditions.
+  checkRequirementConditions(requiredConditions: [string]) {
```

Comment:
nit: i think this type should technically be `string[]`? i think the `[type]` is for tuples

Reply:
Oops, this was my early TypeScript.  Will fix!